share/functions/fish_fossil_prompt.fish

```diff
@@ -0,0 +1,53 @@
+function fish_fossil_prompt --description 'Write out the fossil prompt'
+    # Bail if fossil is not available
+    if not command -sq fossil
+        return 1
+    end
+
+    # Bail if not a fossil checkout
+    if not test -e .fslckout
```

Comment:
This is too strict - like git and hg looking for a .git or .hg directory, fossil looks in parent directories as well - see https://fossil-scm.org/home/doc/tip/www/tech_overview.wiki:

> For Fossil commands that run from within a working checkout, the first thing that happens is that Fossil locates the checkout database. Fossil first looks in the current directory. If not found there, it looks in the parent directory. If not found there, the parent of the parent. And so forth until either the checkout database is found or the search reaches the root of the file system. (In the latter case, Fossil returns an error, of course.) Once the checkout database is located, it is used to locate the repository database. 

So you need to basically do what `fish_print_hg_root` does, except replacing `.hg/dirstate` with `.fslckout` (and *possibly* `_FOSSIL_` - no idea when that name is used).

Alternatively, if `fossil info` is quick enough outside of a repo, say ~5ms (like git), you can remove this check and do

```fish
set -l fossil_info (fossil info)
or return
```

below (possibly with `fossil info 2>/dev/null` if it writes an error).

Reply:
From what I can find `_FOSSIL_` was the old name, and it's been renamed before 2016. So it's fine to ignore for our use, if we need to check for the checkout database ourselves.

Reply:
changed